### PR TITLE
Merge 'config-proxy' and 'proxy ssh' commands logic

### PR DESCRIPTION
**Pull request by smallinsky**
## What:

* Update `tsh config` template and replace TLS Routing unaware `tsh config-proxy` command by `tsh proxy ssh` command: 
Before: 
  ```
  Host *.ice-berg.dev !ice-berg.dev
      Port 3022
      ProxyCommand "/Users/marek/Go/bin/tsh" config-proxy --login="%r" --proxy=ice-berg.dev:3023 %h:%p "ice-berg.dev"
   ```
  
  After: 
  ```
  Host *.ice-berg.dev !ice-berg.dev
      Port 3022
      ProxyCommand "/Users/marek/Go/bin/tsh" proxy ssh --cluster=ice-berg.dev --proxy=ice-berg.dev %r@%h:%p
  ```
* Based on https://github.com/gravitational/teleport/pull/7848 replace ssh agent with identity file login in `tsh proxy ssh` logic

### CODE REVIEW

File: tool/tsh/config.go [hunk 1]
```diff
@@ -41,7 +42,7 @@ Host *.{{ .ClusterName }} {{ .ProxyHost }}
 # Flags for all {{ .ClusterName }} hosts except the proxy
 Host *.{{ .ClusterName }} !{{ .ProxyHost }}
     Port 3022
-    ProxyCommand "{{ .TSHPath }}" config-proxy --login="%r" --proxy={{ .ProxyHost }}:{{ .ProxyPort }} %h:%p "{{ .ClusterName }}"
+    ProxyCommand "{{ .TSHPath }}" proxy ssh --cluster={{ .ClusterName }} --proxy={{ .ProxyHost }} %r@%h:%p
```
Comment (r0mant) [inline tool/tsh/config.go:45]: Will this work if `proxy_listener_mode` is not `multiplex`?

Also, why is `ProxyPort` no longer used?

File: tool/tsh/tsh.go [hunk 2]
```diff
@@ -536,6 +536,7 @@ func Run(args []string, opts ...cliOption) error {
 	// config-proxy is a wrapper to ensure Windows clients can properly use
 	// `tsh config`. As it's not intended to run by users directly and may
 	// not have a stable CLI interface, hide it.
+	// [Deprecated] The config-proxy logic was moved to the tsh proxy ssh command.
```
Comment (r0mant) [inline tool/tsh/tsh.go:539]:
```suggestion
	// DELETE IN 9.0: The config-proxy logic was moved to the tsh proxy ssh command.
```

**Review by r0mant — COMMENTED** [review]
@timothyb89 Could you take a look too please since you worked on `proxy-config` command?

File: tool/tsh/config.go (hunk 1 above)
Comment (smallinsky) [reply tool/tsh/config.go:45]: > Will this work if proxy_listener_mode is not multiplex?

yes, `tsh proxy ssh` logic  distinguish the `proxy_listener_mode` https://github.com/gravitational/teleport/pull/8920/files#diff-2ce402ac3a1c6b5da58e90affa61ed55008e021f83e2ba5f438fc7ec6af95e9aR57

> Also, why is ProxyPort no longer used?

`tsh proxy ssh`  obtains proxy configuration from the `~/.tsh/` proxy profile  so the port address is not longer needed.

File: lib/srv/alpnproxy/local_proxy.go [hunk 3]
```diff
@@ -124,14 +122,13 @@ func (l *LocalProxy) SSHProxy() error {
 	}
 	defer upstreamConn.Close()
 
-	sshAgent, err := getAgent()
 	if err != nil {
 		return trace.Wrap(err)
 	}
```
[the remark this answers is not in the document]
Comment (smallinsky) [reply lib/srv/alpnproxy/local_proxy.go]: Right, I have missed that. Thanks!

**Review by r0mant — APPROVED** [review]

File: tool/tsh/config.go (hunk 1 above)
Comment (timothyb89) [reply tool/tsh/config.go:45]: Really glad to see the port requirement go away!

**Review by timothyb89 — APPROVED** [review]
Looks great! TLS routing, SSH fallback, and backwards compat w/ v7 config are all working for me.

**Pull request merged**